.github/workflows/deploy-to-gcs.yml

```diff
@@ -0,0 +1,50 @@
+name: Deploy repository to GCS
+
+on:
+  workflow_call:
+    inputs:
+      gcp_workload_identity_provider:
+        required: true
+        type: string
+      gcp_service_account:
+        required: true
+        type: string
+
+permissions: {}
+
+jobs:
+  deploy-to-gcs:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: 'write' # For authenticating with the GitHub workflow identity
+
+    steps:
+      - uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe # v4.1.2
+        with:
+          name: github-pages
+
+      - name: Prepare data for upload
+        run: |
+          # Extract the github-pages arcive into ./repository/
+          mkdir repository
+          tar --directory repository -xvf artifact.tar
+
+      - uses: google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f # v2.1.1
+        with:
+          token_format: access_token
+          workload_identity_provider: ${{ inputs.gcp_workload_identity_provider }}
```

Comment:
Do you think we need these as parameters? Given that we hardcode the project ID and the bucket name, this isn't usable in different environments. Other option is to parameterize everything so this could be reused across deployments.

Reply:
The minor advantage of this setup is that it avoids defining the service account and provider in two separate places (since they're in the repository variables already). I'm ok with hard coding as well.

If this looks like it might be reusable, we can consider moving it to tuf-on-ci itself and making it properly parameterized (but I have doubts: this might always be a bit sigstore-specific).

Reply:
It might be useful within tuf-on-ci, though given it's specific to GCP and we don't have a need to support this on other cloud platforms at this point, leaving it within this repo for now sgtm.